integrationtests/model_storage/integrationtest_model_storage.py

```diff
@@ -0,0 +1,164 @@
+import pathlib
+import shutil
+
+import grpc
+from integrationtests.utils import get_modyn_config
+from modyn.common.ftp import delete_file, download_file, upload_file
+from modyn.metadata_database.metadata_database_connection import MetadataDatabaseConnection
+from modyn.metadata_database.models import TrainedModel, Trigger
+from modyn.model_storage.internal.grpc.generated.model_storage_pb2 import (
+    DeleteModelRequest,
+    DeleteModelResponse,
+    FetchModelRequest,
+    FetchModelResponse,
+    RegisterModelRequest,
+    RegisterModelResponse,
+)
+from modyn.model_storage.internal.grpc.generated.model_storage_pb2_grpc import ModelStorageStub
+from modyn.utils import grpc_connection_established
+
+TEST_MODELS_PATH = pathlib.Path("/app") / "model_storage" / "test_models"
+TEST_FILE_NAME_LOCAL = "test_model_local.txt"
+TEST_FILE_NAME_LOCAL_RESP = "test_model_local_response.txt"
+TEST_FILE_NAME_REMOTE = "test_model_remote.txt"
+
+
+def create_dummy_file():
+    pathlib.Path(TEST_MODELS_PATH).mkdir(parents=True, exist_ok=True)
+
+    with open(TEST_MODELS_PATH / TEST_FILE_NAME_LOCAL, "w") as f:
+        f.write("Test model storage component")
+
+
+def cleanup_models_dir() -> None:
+    shutil.rmtree(TEST_MODELS_PATH)
+
+
+def connect_to_model_storage(config: dict) -> grpc.Channel:
+    model_storage_address = f"{config['model_storage']['hostname']}:{config['model_storage']['port']}"
+    model_storage_channel = grpc.insecure_channel(model_storage_address)
+
+    if not grpc_connection_established(model_storage_channel) or model_storage_channel is None:
+        raise ConnectionError(f"Could not establish gRPC connection to component at {model_storage_address}.")
+
+    return model_storage_channel
+
+
+def upload_dummy_file_to_trainer(config: dict):
+    upload_file(
+        config["trainer_server"]["hostname"],
+        int(config["trainer_server"]["ftp_port"]),
+        "modyn",
+        "modyn",
+        local_file_path=TEST_MODELS_PATH / TEST_FILE_NAME_LOCAL,
+        remote_file_path=pathlib.Path(TEST_FILE_NAME_REMOTE),
+    )
+
+
+def delete_dummy_file_from_trainer(config: dict):
+    delete_file(
+        config["trainer_server"]["hostname"],
+        int(config["trainer_server"]["ftp_port"]),
+        "modyn",
+        "modyn",
+        pathlib.Path(TEST_FILE_NAME_REMOTE),
+    )
+
+
+def insert_trigger_into_database(config: dict) -> (int, int):
+    with MetadataDatabaseConnection(config) as database:
+        pipeline_id = database.register_pipeline(2)
+
+        trigger = Trigger(trigger_id=10, pipeline_id=pipeline_id)
+        database.session.add(trigger)
+        database.session.commit()
+
+        return trigger.pipeline_id, trigger.trigger_id
+
+
+def delete_data_from_database(config: dict, pipeline_id: int, trigger_id: int, model_id: int):
+    with MetadataDatabaseConnection(config) as database:
+        database.session.query(TrainedModel).filter(TrainedModel.model_id == model_id).delete()
```

Comment:
This should be done by the delete request, shouldn't it? I wonder why this does not fail. This indicates to me that something silently fails right now, because if the grpc component would delete the model, then we should not be able to do it again here I think. Can you investigate why this works and find the potential problem? And then we can remove this line here

Reply:
You are right about this line - it is unnecessary.
Maybe I am missing something about the semantics of sqlalchemy but in this case we are filtering by a specific (non-present) model id resulting in an empty result. Hence, there is nothing to delete and no error is thrown.

Reply:
ahhh interesting - thanks for investigating!